oprf/oprf.go

```diff
@@ -139,24 +146,32 @@ func generateContext(id SuiteID) (ctx [3]byte) {
 	return
 }
 
-// GenerateKeyPair generates a KeyPair in accordance with the group.
-func GenerateKeyPair(s *suite) KeyPair {
+func (s *suite) generateKeyPair() *KeyPair {
 	r := s.Random()
 	privateKey := r.RndScl(rand.Reader)
 	publicKey := s.NewElt()
 	publicKey.MulGen(privateKey)
 
-	return KeyPair{publicKey, privateKey}
+	return &KeyPair{s.SuiteID, publicKey, privateKey}
+}
+
+// GenerateKeyPair generates a KeyPair in accordance with the group.
+func GenerateKeyPair(id SuiteID) (*KeyPair, error) {
+	suite, err := suiteFromID(id)
+	if err != nil {
+		return nil, err
+	}
+	return suite.generateKeyPair(), nil
 }
 
 func suiteFromID(id SuiteID) (*suite, error) {
 	switch id {
 	case OPRFP256:
-		return &suite{group.P256, crypto.SHA256}, nil
+		return &suite{id, group.P256, crypto.SHA256}, nil
```

Comment:
Mmm.. why do we need to pass the id if we already know the group and the hash?

Reply:
it's carried to generate context_string later.